cloudevents/working-drafts/xml-format.md

```diff
@@ -0,0 +1,243 @@
+adhere to, and align with, the conventions in defined in the [cloud event specification][ce-spec].
+
+See the [envelope](#4-envelope) for special handing of the `specversion` context attribute.
+
+Extension attributes SHOULD be expressed with the appropriate `xsi:type` to allow them
+to be exchanged without loss of type information. Extension attributes with no `xsi:type`
+discriminator SHOULD be interpreted with an implied type of `xs:string`
+
+``` xml
+    ...
+    <id>AAABBBCCCNNN0000</id>
+    ..
+    <time>2021-08-14T-08:00</time>
+    ..
+    <myextension xsi:type="xs:string">my extension value</myextension>
+    <myboolean xsi:type="xs:boolean">false</myboolean>
+```
+
+## 3. Data
+
+The `Data` portion of a CloudEvent follows a similar model to that employed by
+the [JSON Format specification][json-format]. The element names MUST be capitalized to avoid
+collision with curent or future CloudEvent attribute names.
+
+### 3.1 Binary Data
+
+MUST be carried in an element with an defined type of `xs:base64Binary`
+
+``` xml
+<Data xsi:type="xs:base64Binary">.........</Data>
+```
+
+### 3.2 Text Data
+
+MUST be carried in an element with an defined type of `xs:string`
+
+``` xml
+<Data xsi:type="xs:string">This is text</Data>
+```
+
+### 3.3 XML Data
+
+XML data MUST be carried in an explicit element `<DataXml>`:
+
+``` xml
+<DataXml>
+        <myData>
+            ....
+        </myData>  
+</DataXml>
+```
+
+## 4. Envelope
+
+Each CloudEvent is wholly represented as an XML element, the root xml
+document element MUST be `<CloudEvent>`. This element carries the `specversion`
+as an XML attribute value.
+
+Such a representation MUST use the media type `application/cloudevents+xml`.
+
+The enveloping element contains:
+
+* A set of context attributre elements.
+* Optionally: Either `<Data>` or `<DataXml>`
+
+eg (Namespace definitions omitted for brevity):
+
+``` xml
+<CloudEvent specversion="1.0">
+    <time>2020-03-19T12:54:00-07:00</time>
+    <datacontenttype>text/plain</datacontenttype>
+    <id>000-1111-2222</id>
+    <source>urn:uuid:123e4567-e89b-12d3-a456-426614174000</source>
+    <type>SOME.EVENT.TYPE</type>
+    <myboolean xsi:type="xs:boolean">false</myboolean>
+    <Data xsi:type="xs:string">Now is the winter of our discount tents...</Data>
+</CloudEvent>
+```
+
+## 5. XML Batch Format
+
+In the _XML Batch Format_ several CloudEvents are batched into a single XML
+document. The document comprises a list of elements in the XML Format.
+
+Although the _XML Batch Format_ builds on top of the _XML Format_, it is
+considered as a separate format: a valid implementation of the _XML Format_
+doesn't need to support it. The _XML Batch Format_ MUST NOT be used when only
+support for the _XML Format_ is indicated.
+
+An XML Batch of CloudEvents MUST use the media type
+`application/cloudevents-batch+xml`.
+
+Example:
+
+``` xml
+<CloudEventBatch>
+    <CloudEvent specversion="1.0">
```

Comment:
I'm sure in *some* spec there's something about all events in the same batch having to use the same specversion. I can't find it now though. (I've always disagreed with it anyway...) Not sure whether it needs calling out at all - worth a 2-minute discussion in the meeting (where others will no doubt be able to find the rule I'm remembering badly...)

Reply:
Ignore this - it's in the HTTP transport. No need to put anything in here.